Code/Framework/AzCore/Tests/Memory/PoolAllocatorTests.cpp

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#include <AzCore/UnitTest/TestTypes.h>
```

Comment:
We actually have PoolAllocatorTest in Memory.cpp https://github.com/o3de/o3de/pull/14824/files#diff-66fc5113ad002d6ac22403c3f7aaa33d8b2d6050e10b61dc386f72eb0d8bffa7R669-R676